utils/text_formatting.tsx

```diff
@@ -343,25 +337,8 @@ export function sanitizeHtml(text: string): string {
 }
 
 // Copied from our fork of commonmark.js
-const emailAlphaNumericChars = '\\p{L}\\p{Nd}';
-const emailSpecialCharacters = "!#$%&'*+\\-\\/=?^_`{|}~";
-const emailRestrictedSpecialCharacters = '\\s(),:;<>@\\[\\]';
-const emailValidCharacters = emailAlphaNumericChars + emailSpecialCharacters;
-const emailValidRestrictedCharacters = emailValidCharacters + emailRestrictedSpecialCharacters;
-const emailStartPattern =
-  '(?:[' +
-  emailValidCharacters +
-  '](?:[' +
-  emailValidCharacters +
-  ']|\\.(?!\\.|@))*|\\"[' +
-  emailValidRestrictedCharacters +
-  '.]+\\")@';
-const reEmail = XRegExp.cache(
-    '(^|[^\\pL\\d])(' +
-    emailStartPattern +
-    '[\\pL\\d.\\-]+[.]\\pL{2,4}(?=$|[^\\p{L}]))',
-    'g',
-);
+// eslint-disable-next-line no-useless-escape
+const emailRegex = /(^|[^\p{L}\d])((?:[\p{L}\p{Nd}!#$%&'*+-/=?^_`\{|\}~](?:[\p{L}\p{Nd}!#$%&'*+-/=?^_`\{|\}~]|.(?!.|@))*|"[\p{L}\p{Nd}!#$%&'*+-/=?^_`\{|\}~\s(),:;<>@[].\]+")@[\p{L}\d.-]+[.]\p{L}{2,4}(?=$|[^\p{L}]))/gu;
```

Comment:
@hmhealey this regex is failing for one case where `email.@domain.com` could you please check it out

Reply:
Wow, that regex looks extremely complex!

Reply:
yes thats because of so many escapes need in unicode flag https://github.com/mattermost/mattermost-webapp/pull/9805#discussion_r805305369

Reply:
@M-ZubairAhmed this one is not in any way less complex, but it checks (fails) for the dot problem you mentioned: https://regex101.com/r/sI6yF5/1

I have used that before and it works like a charm.

Reply:
@michelengelen the one you mentioned fails for `email.@domain.com` i wonder if we can modify the regex to catch that also. I am a noob in regex

Reply:
Before changing to a simpler regex, it would be nice to watch over the history of changes to know why this become like this. Maybe there are some corner cases that we wanted to address with this, and we would need to make sure any simplification still cover those cases.

Reply:
@larkox absolutely ... It is just that the regex that has been used is not very performant and the first capture group does not do anything at all

Reply:
Good news i was able to fix the existing regex to pass the tests

Reply:
I'm pretty sure that first group was supposed to be a word boundary, but I couldn't use `\b` because a URL could start with something that wasn't considered to be a word character. It might be that Unicode letters aren't recognized as word characters so `\b` doesn't work for them.

If emails consistently started with a non-word character, then we could use `\B` like you'll see in the emoji and at mention parsing, but since it can start with either word or non-word characters, it means we have to do some uglier stuff like in this regex